gemrb/core/Map.cpp

```diff
@@ -2397,12 +2397,12 @@ void Map::PlayAreaSong(int SongType, bool restart, bool hard) const
 	// Some subareas don't have their own songlist. It is currently unclear
 	// how the different games handle this situation and which music GemRB
 	// should play, if any. Further research is needed.
-	// At least for the battle music in BG1, e.g. AR2607 (intro candlekeep
-	// ambush south), there is a strong assumption to play the music from
-	// the masterarea's songlist.
+	// At least for BG1 there is a strong assumption to play the music from
+	// the masterarea's songlist, e.g. AR2607 (intro candlekeep ambush south),
+	// or AR2302 (friendly arm inn 2nd floor).
```

Comment:
Is this really still true? The current working hypothesis is that the last worldmap area is used.

Reply:
Haven't had the time to test this further unfortunately.

Reply:
It's easy to test:
- pick and area with subsequent subareas (eg. the store in BG, some tower, deep dungeon)
- make the first subarea a master area by adding it to mastarea.2da
- ensure the first two have different songs set
- test what happens in the third area